modules/alloy/src/test/ts/browser/behaviour/BlockingTest.ts

```diff
@@ -0,0 +1,115 @@
+import { Chain, Log, Step, UiFinder } from '@ephox/agar';
+import { Assert, UnitTest } from '@ephox/bedrock-client';
+import { Optional, OptionalInstances } from '@ephox/katamari';
+import { Attribute, SugarElement } from '@ephox/sugar';
+import * as Behaviour from 'ephox/alloy/api/behaviour/Behaviour';
+import { Blocking } from 'ephox/alloy/api/behaviour/Blocking';
+import { Replacing } from 'ephox/alloy/api/behaviour/Replacing';
+import * as GuiFactory from 'ephox/alloy/api/component/GuiFactory';
+import { AlloySpec } from 'ephox/alloy/api/component/SpecTypes';
+import * as Memento from 'ephox/alloy/api/component/Memento';
+import * as GuiSetup from 'ephox/alloy/api/testhelpers/GuiSetup';
+import { TestStore } from 'ephox/alloy/api/testhelpers/TestHelpers';
+
+UnitTest.asyncTest('BlockingTest', (success, failure) => {
+  const blockRoot = Memento.record({
+    dom: {
+      tag: 'div'
+    },
+    behaviours: Behaviour.derive([
+      Replacing.config({})
+    ])
+  });
+
+  const makeComponent = (store: TestStore, _doc: SugarElement<Document>, _body: SugarElement<HTMLBodyElement>) => {
+    const component = GuiFactory.build({
+      dom: {
+        tag: 'div'
+      },
+      components: [
+        blockRoot.asSpec()
+      ],
+      behaviours: Behaviour.derive([
+        Blocking.config({
+          getRoot: () => blockRoot.getOpt(component),
+          onBlock: store.adder('block'),
+          onUnblock: store.adder('unblock')
+        })
+      ])
+    });
+
+    return component;
+  };
+
+  GuiSetup.setup(makeComponent, (doc, body, gui, comp, store) => {
+    const sBlock = (fn?: (behaviour: Record<string, Behaviour.ConfiguredBehaviour<any, any>>) => AlloySpec) => Step.sync(() => Blocking.block(comp, fn));
+    const sUnblock = Step.sync(() => Blocking.unblock(comp));
+
+    return [
+      Log.stepsAsStep('TINY-6487', 'Blocking events are fired', [
```

Comment:
Could we reword this to explain that the `onBlock` and `onUnblock` handlers are run, rather than saying `events`?